src/Kiota.Builder/Plugins/PluginsGenerationService.cs

```diff
@@ -93,8 +97,91 @@ public async Task GenerateManifestAsync(CancellationToken cancellationToken = de
             }
             await writer.FlushAsync(cancellationToken).ConfigureAwait(false);
         }
+
+        // 3. write the app manifest if its an Api Plugin
+        if (Configuration.PluginTypes.Any(static plugin => plugin == PluginType.APIPlugin))
+        {
+            var manifestFullPath = Path.Combine(Configuration.OutputPath, AppManifestFileName);
+            var pluginFileName = $"{Configuration.ClientClassName.ToLowerInvariant()}-{PluginType.APIPlugin.ToString().ToLowerInvariant()}{ManifestFileNameSuffix}";
+            var appManifestModel = await GetAppManifestModelAsync(pluginFileName, manifestFullPath, cancellationToken).ConfigureAwait(false);
+#pragma warning disable CA2007
+            await using var appManifestStream = File.Open(manifestFullPath, FileMode.Create);
+#pragma warning restore CA2007
+            await JsonSerializer.SerializeAsync(appManifestStream, appManifestModel, AppManifestModelGenerationContext.AppManifestModel, cancellationToken).ConfigureAwait(false);
+        }
+    }
+
+    private async Task<AppManifestModel> GetAppManifestModelAsync(string pluginFileName, string manifestFullPath, CancellationToken cancellationToken)
+    {
+        var manifestInfo = ExtractInfoFromDocument(OAIDocument.Info);
+        // create default model
+        var manifestModel = new AppManifestModel
+        {
+            Schema = "https://developer.microsoft.com/json-schemas/teams/vDevPreview/MicrosoftTeams.schema.json",
+            ManifestVersion = "devPreview",
+            Version = "1.0.0",
+            Id = Guid.NewGuid().ToString(),
+            Developer = new Developer
+            {
+                Name = !string.IsNullOrEmpty(OAIDocument.Info?.Contact?.Name) ? OAIDocument.Info?.Contact?.Name : "Kiota Generator, Inc.",
```

Comment:
```suggestion
                Name = !string.IsNullOrEmpty(OAIDocument.Info?.Contact?.Name) ? OAIDocument.Info?.Contact?.Name : "Microsoft kiota",
```
Did you want to start accepting payments? 🤣

Reply:
Haha. I was just copying these from the spec. https://github.com/microsoft/kiota/blob/main/specs/cli/plugin-add.md#app-manifest

Will update in a bit.